src/Ledger/Certs.lagda

```diff
@@ -264,42 +264,41 @@ constitutional committee.
 \end{itemize}
 
 \begin{figure*}[h]
-\begin{AgdaSuppressSpace}
+\begin{AgdaMultiCode}
 \begin{code}[hide]
 data
 \end{code}
 \begin{code}
-  _⊢_⇀⦇_,DELEG⦈_     : DelegEnv → DState → DCert → DState → Type
+    _⊢_⇀⦇_,DELEG⦈_     : DelegEnv → DState → DCert → DState → Type
 \end{code}
 \begin{code}[hide]
 data
 \end{code}
 \begin{code}
-  _⊢_⇀⦇_,POOL⦈_      : PoolEnv → PState → DCert → PState → Type
+    _⊢_⇀⦇_,POOL⦈_      : PoolEnv → PState → DCert → PState → Type
 \end{code}
 \begin{code}[hide]
 data
 \end{code}
 \begin{code}
-  _⊢_⇀⦇_,GOVCERT⦈_   : GovCertEnv → GState → DCert → GState → Type
+    _⊢_⇀⦇_,GOVCERT⦈_   : GovCertEnv → GState → DCert → GState → Type
 \end{code}
 \begin{code}[hide]
 data
 \end{code}
 \begin{code}
-  _⊢_⇀⦇_,CERT⦈_      : CertEnv → CertState → DCert → CertState → Type
+    _⊢_⇀⦇_,CERT⦈_      : CertEnv → CertState → DCert → CertState → Type
 \end{code}
 \begin{code}[hide]
 data
 \end{code}
 \begin{code}
-  _⊢_⇀⦇_,CERTBASE⦈_  : CertEnv → CertState → ⊤ → CertState → Type
-\end{code}
-\begin{code}
-_⊢_⇀⦇_,CERTS⦈_     : CertEnv → CertState → List DCert → CertState → Type
-_⊢_⇀⦇_,CERTS⦈_ = ReflexiveTransitiveClosureᵇ _⊢_⇀⦇_,CERTBASE⦈_ _⊢_⇀⦇_,CERT⦈_
+    _⊢_⇀⦇_,CERTBASE⦈_  : CertEnv → CertState → ⊤ → CertState → Type
+
+_⊢_⇀⦇_,CERTS⦈_       : CertEnv → CertState → List DCert → CertState → Type
+_⊢_⇀⦇_,CERTS⦈_ = ReflexiveTransitiveClosure _⊢_⇀⦇_,CERT⦈_
```

Comment:
It would be much nicer if you just make an abstraction here that replaces `ReflexiveTransitiveClosureᵇ`. Then you wouldn't have to modify `LEDGER` at all, which is better because it keeps the concerns of the different STSs separate.

Reply:
It also means that the proof that this new closure operator preserves `Computational` would be general, whereas right now it's intertwined with the `Computational` proof of `LEDGER`.

Reply:
@WhatisRT I've tried a number of ways to replace `ReflexiveTransitiveClosureᵇ` with an abstraction that, in this instance, does `CERTBASE` for the base case and then `CERTS` for the `sigs` case, while also not having the empty list appear in the `CERTBASE` case... but I'm not able to get the completeness proofs to work with the various attempts I've made.  Could you be more specific about what you had in mind by "just make an abstraction"?

Reply:
Just do exactly what you did, except in general:
```
module _ (_⊢_⇀⟦_⟧ᵇ_ : C → S → ⊤ → S → Type) (_⊢_⇀⟦_⟧_ : C → S → Sig → S → Type) where
  data ReflexiveTransitiveClosureᵇ' : C → S → List Sig → S → Type where
      RTC :
          ∙ Γ ⊢ s ⇀⟦ _ ⟧ᵇ s'
          ∙ ReflexiveTransitiveClosure _⊢_⇀⟦_⟧_ Γ s' sigs s''
          ───────────────────────────────────────
          ReflexiveTransitiveClosureᵇ' Γ s sigs s''
```
Then, whatever was necessary to prove `Computational` for `LEDGER` also applies to this relation, and all you need to do is keep the original definitions of `CERTS` and `LEDGER`, but just use this closure instead. If it turns out that the original `ReflexiveTransitiveClosureᵇ` is unused, it's best to just call this relation `ReflexiveTransitiveClosureᵇ`.

Reply:
I see.  That makes sense.  Thanks.

I've tried it and I'm still having trouble getting the computation proofs to go through... I'll keep trying...

Reply:
Got em to proved!🍾🎆